Doc/c-api/module.rst

```diff
@@ -449,10 +449,11 @@ state:
    Add an object to *module* as *name*.  This is a convenience function which
    can be used from the module's initialization function.
 
-   On success, return ``0``. On error, raise an exception and return ``-1``.
```

Comment:
Please keep this general statement. There are many cases where the function can fail. For example, if mod is not a module or has no dictionary. PyDict_SetItemString() can fail with MemoryError, etc.